Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -162,29 +162,53 @@ void APSStarActor::OnInitialized(const FVector& PreviousActorLocation)
+	if (!ensureMsgf(StarMeshComponent, TEXT("ASSERT: [%i] %hs:\n'StarMeshComponent' is not valid!"), __LINE__, __FUNCTION__))
 	{
 		return; // Early return if pointers are invalid
 	}
 
+	UMaterialInstanceDynamic* StarProgressionDynamicMaterial = nullptr;
+
 	// locked stars
 	if (StarActorState == EPSStarActorState::Locked)
 	{
-		StarMeshComponent->SetMaterial(0, UPSDataAsset::Get().GetLockedProgressionMaterial());
+		const float LockedStar = 0.0f;
+		StarProgressionDynamicMaterial = UPSWorldSubsystem::Get().GetStarProgressionDynamicMaterial(EPSStarActorState::Locked);
+		SetStarActorProgressMeshMaterial(StarProgressionDynamicMaterial, LockedStar);
 		return;
 	}
 
 	// unlocked stars with fractional part
 	if (AmountOfStars > 0 && AmountOfStars < 1)
 	{
-		StarMeshComponent->SetMaterial(0, StarDynamicProgressMaterial);
-		StarDynamicProgressMaterial->SetScalarParameterValue(UPSDataAsset::Get().GetStarMaterialSlotName(), AmountOfStars / UPSDataAsset::Get().GetStarMaterialFractionalDivisor()); // StarMaterialFractionalDivisor is hardcoded value to 3 to tweak bad UV to simulate it's working
+		const float PartialUnLockedStarAmount = AmountOfStars / UPSDataAsset::Get().GetStarMaterialFractionalDivisor();
+		StarProgressionDynamicMaterial = UPSWorldSubsystem::Get().GetStarProgressionDynamicMaterial(EPSStarActorState::Partial); // StarMaterialFractionalDivisor is hardcoded value to 3 to tweak bad UV to simulate it's working
+		SetStarActorProgressMeshMaterial(StarProgressionDynamicMaterial, PartialUnLockedStarAmount);
 		return; // Early return for fractional stars
 	}
 
 	// unlocked stars EPSStarActorState::Unlocked
-	StarMeshComponent->SetMaterial(0, UPSDataAsset::Get().GetUnlockedProgressionMaterial());
+	const float UnLockedStarAmount = 1.0f;
+	StarProgressionDynamicMaterial = UPSWorldSubsystem::Get().GetStarProgressionDynamicMaterial(EPSStarActorState::Unlocked);
+	SetStarActorProgressMeshMaterial(StarProgressionDynamicMaterial, UnLockedStarAmount);
+}
+
+// Applies the star dynamic material
+void APSStarActor::SetStarActorProgressMeshMaterial(class UMaterialInstanceDynamic* StarDynamicMaterial, float StarProgressionAmount)
+{
+	if (!ensureMsgf(StarDynamicMaterial, TEXT("ASSERT: [%i] %hs:\n'StarDynamicMaterial' is not valid!"), __LINE__, __FUNCTION__))
+	{
+		return; // Early return if pointers are invalid
+	}
+
+	// Add a face texture over current star material
+	const FPSRowData& CurrentSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();
+	const FName StarFaceTextureParameter = UPSDataAsset::Get().GetStarFaceTextureParameter();
+	UTexture* StarFaceTexture = CurrentSettingsRowData.StarFaceTexture;
+
+	StarMeshComponent->SetMaterial(0, StarDynamicMaterial);
+	StarDynamicMaterial->SetScalarParameterValue(UPSDataAsset::Get().GetStarMaterialSlotName(), StarProgressionAmount);
+	StarDynamicMaterial->SetTextureParameterValue(StarFaceTextureParameter, StarFaceTexture);
```

Comment:
Next two lines look incosistent:
```
#1	StarDynamicMaterial->SetScalarParameterValue(UPSDataAsset::Get().GetStarMaterialSlotName(), StarProgressionAmount);
#2	StarDynamicMaterial->SetTextureParameterValue(StarFaceTextureParameter, StarFaceTexture);
```
, where line 1 you provide parameter name from the Data Asset directly, but line 2 it's taken from local variable. Please choose 1 approach, without mixing

Reply:
i'd like to use local variable so make it consistent https://github.com/moinm3uw/ProgressionSystem/pull/55/commits/2d4d7467a88cad6c4b4e1a678ae199b720c329d2
let me know if I should use directly getter